# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1276570399 25200
#      Mon Jun 14 19:53:19 2010 -0700
# Node ID d500acdcf2dde0dc7dab137488c1eb47a21ab7d8
# Parent  0bcd17cad9d134d2db6f3ce7a259755afc1c2e54
Remove extra newlines when displaying multi-dimensional arrays.
Fixes bug #30029.

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,13 @@
+2010-06-14  Rik <octave@nomad.inbox5.com>
+
+	* pr-output.cc: Remove extra newlines when displaying multi-dimensional
+        arrays (bug #30029).
+
 2010-06-14  Jaroslav Hajek  <highegg@gmail.com>
 
 	* data.cc (Fsort, Fnth_element): Fix behavior when trailing singleton
 	dim is specified.
 
 2010-06-12  Ben Abbott <bpabbott@mac.com>
 
 	* DLD-FUNCTIONS/gammainc.cc: Fix doc-string typo.
diff --git a/src/pr-output.cc b/src/pr-output.cc
--- a/src/pr-output.cc
+++ b/src/pr-output.cc
@@ -1890,18 +1890,26 @@ octave_print_internal (std::ostream& os,
               idx(1) = idx_vector (':'); \
  \
               for (int k = 2; k < ndims; k++) \
                 idx(k) = idx_vector (ra_idx(k)); \
  \
               octave_value page \
                 = MAT_T (Array<ELT_T> (nda.index (idx), nr, nc)); \
  \
-              page.print_with_name (os, nm); \
- \
+              if (i < ndims - 2) \
+                { \
+                  page.print_with_name (os, nm); \
+                } \
+              else \
+                { \
+                  page.print_name_tag (os, nm); \
+                  page.print_raw(os); \
+                } \
+              \
               if (i < m) \
                 NDA_T::increment_index (ra_idx, dims, 2); \
             } \
         } \
     } \
   while (0)
 
 void
